CharlieBackend.Business/Services/CurrentUserService.cs

```diff
@@ -0,0 +1,126 @@
+﻿using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core.Entities;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Security.Claims;
+
+namespace CharlieBackend.Business.Services
+{
+    /// <summary>
+    /// Class that provides access to properties of the current user.
+    /// </summary>
+    public class CurrentUserService : ICurrentUserService
+    {
+        long _accountId;
+        long _entityId;
+        string _email;
+
+        readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CurrentUserService(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public long AccountId
+        {
+            get
+            {
+                if (_accountId == default)
+                {
+                    string accountIdString = _httpContextAccessor
+                        .HttpContext
+                        .User
+                        ?.Claims
+                        ?.SingleOrDefault(claim => claim.Type == "AccountId")
+                        ?.Value;
+
+                    if (!long.TryParse(accountIdString, out _accountId))
+                    {
+                        throw new UnauthorizedAccessException("Not authorized!");
+                    }
+                }
+
+                return _accountId;
+            }
+        }
+
+        /// <summary>
+        /// Property that provides access to entity id of the current user.
+        /// Corresponds to one of the following entities: 
+        /// <list type="bullet">
+        /// <item><see cref="Student"/></item>
+        /// <item><see cref="Secretary"/></item>
+        /// <item><see cref="Mentor"/></item>
+        /// </list>
+        /// </summary>
+        public long EntityId
+        {
+            get
+            {
+                if (_entityId == default)
+                {
+                    string entityIdString = _httpContextAccessor
+                        .HttpContext
+                        .User
+                        ?.Claims
+                        ?.SingleOrDefault(claim => claim.Type == "Id")
+                        ?.Value;
+
+                    if (!long.TryParse(entityIdString, out _entityId))
+                    {
+                        throw new UnauthorizedAccessException("Not authorized!");
+                    }
+                }
+
+                return _entityId;
+            }
+        }
+
+        public string Email
+        {
+            get
+            {
+                if (_email is null)
+                {
+                    _email = _httpContextAccessor
```

Comment:
Consider to create a private method to return appropriate claim and substitute common logic.

Reply:
Fixed. Moved common getters' common logic of obtaining claim value into a private method.
In this commit: https://github.com/ita-social-projects/WhatBackend/pull/349/commits/3bf40030b640ffccad4698bb3f5bd75f6616c5ed